### ISSUE TITLE
Problem with hydrate entity after update to 2.1.5

### ISSUE THREAD

**Issue by werwolf666** [doctrine/DoctrineModule#653]
I have entity with ManyToOne relationships:
class VehicleParametrized
{
~~~~~~~~~~~~~~
    /**
     * @var Vehicle
     * @ManyToOne(targetEntity="DM\Entity\Common\Vehicle")
     * @JoinColumns({
     *   @JoinColumn(name="vehicle_id", referencedColumnName="id",
     *     onDelete="CASCADE", nullable=false)
     * })
     */
    private $vehicle = null;

    /**
     * @var FitmentParametersSet
     * @ManyToOne(targetEntity="DM\Entity\Common\FitmentParametersSet")
     * @JoinColumns({
     *   @JoinColumn(name="parameters_set_id", referencedColumnName="id",
     *     onDelete="CASCADE", nullable=true)
     * })
     */
    private $parametersSet = null;
~~~~~~~~~~~~~~~~~~~~~~

I tried to hydrate entity with properties
                        'vehicle' => new Vehicle(),
                        'parametersSet' => null,
and got error 'No mapping found for field 'parametersSet' on class 'DM\Entity\Common\VehicleParametrized'.

In version 2.1.4 this works correctly, but in 2.1.5 was added checking
 if (is_null($value) && method_exists($this->metadata, 'isNullable') && $this->metadata->isNullable($name))

But isNullable works only for fieldMappings property of ClassMetaData object and can't check associationMappings properties.

**Comment by TomHAnderson** [comment]
Should be fixed with #655

**Comment by TomHAnderson** [comment]
Hydrators have been moved to https://github.com/doctrrine/doctrine-laminas-hydrator
